applications/minotari_app_grpc/proto/wallet.proto

```diff
@@ -249,13 +249,23 @@ enum TransactionStatus {
   TRANSACTION_STATUS_COINBASE_NOT_IN_BLOCK_CHAIN = 14;
 }
 
-message GetCompletedTransactionsRequest {}
+message GetCompletedTransactionsRequest {
+  user_payment_id payment_id = 1;
+}
 
 message GetCompletedTransactionsResponse {
   TransactionInfo transaction = 1;
 }
 
-message GetBalanceRequest {}
+message GetBalanceRequest {
+    user_payment_id payment_id = 1;
+}
+
+message user_payment_id {
+  bytes u256 = 1;
+  string utf8_string = 2;
+  bytes user_bytes = 3;
+}
```

Comment:
I think we should only use U256 or string in  this interface. With PaymentId bytes are interpreted as a string.

Reply:
And if your bytes is not a standard utf-8 string?